docs/DeveloperGuide.md

```diff
@@ -2,32 +2,285 @@
+  superclass.
+* The `getNusCourseCode` method then extract out the user specified NUS course code from the input, which is in the
+  format: `filter COURSE_CODE`.
+* The NUS course code is then passed into the `displayMappableCourses()` method along with teh Json object. The method
+  will iterate over the keys of the database which contains the University names, then obtain the array `courses`
+  stored in the "courses" field. The `courses` array is then iterated over, for each course,
+  if the value in the "nus_course_code" is equals to the NUS course code passed into the method, the university name and
+  "pu_course_code" value of the course will be printed to the CLI.
+* There are also assertions and logging in place for error handling.
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML:
+![Filter Courses Sequence Diagram](images/FilterCoursesCommand.png)
+
+### 3. Obtain Partner University Email and Contact Number Command
+
+#### Overview:
+The command is responsible to retrieve the email contact and contact number data for a specified partner
+university. It helps users to reach out to the partner universities for any enquiries about programs or
+exchange opportunities.
+
+#### How the feature is implemented:
+* The `ObtainContactsCommand` class extends `Command` class where it overrides the `execute()` method for
+  custom behaviour.
+* The command first reads a JSON file to obtain the names via `createJsonObject()` method from the
+  superclass.
+* The `getSchoolName()` and `getContactType()` methods are used to parse the user input, extracting the requested 
+  university name and contact type (email or phone number).
+* After parsing, the `findMatchingSchool()` method identifies the correct university entry within the JSON data.
+* The `handleContactType()` method retrieves and prints the requested contact information based on the input, 
+  displaying either the university’s email address or phone number.
+* There are also assertions and logging in place for error handling.
+
+#### Why it is implemented that way:
+* The `execute` method is essential and unique to every command class so inheritance was used.
+* Every method in the class remains maintainable and has one responsibility this allows easy debugging and
+  refactoring.
+* By using inheritance, new command classes can easily extend the functionality of existing ones 
+  which reducing redundancy in the code
+* Logging and assertions helps the team of developers to follow through the command execution.
+
+#### Alternatives considered:
+* Reading of the `database.json` was tricky and other libraries were considered.
+* Considered placing all the class methods inside the `execute` method but kept SLAP in mind to ensure
+  readability.
+
+#### Sequence Diagram on PlantUML:
+![Filter Courses Sequence Diagram](images/ObtainContactsCommand.png)
```

Comment:
Perhaps you could include an [else] label for the first alt block for greater clarity?